router_test.ts

```diff
@@ -0,0 +1,293 @@
+import { createRouter } from "./router.ts";
+import { describe, expect, fn, it } from "./dev_deps.ts";
+import { Status, STATUS_TEXT } from "./deps.ts";
+
+const describeTests = describe("createRouter");
+
+it(
+  describeTests,
+  "should return 404 when url path does not match",
+  async () => {
+    const router = createRouter({});
+    const res = await router(new Request("http://localhost/"));
+
+    expect(res).toEqualResponse(
+      new Response(null, {
+        status: Status.NotFound,
+        statusText: STATUS_TEXT[Status.NotFound],
+      }),
+    );
+  },
+);
+
+it(
+  describeTests,
+  `should return 404 when the handler is not defined`,
+  async () => {
+    const router = createRouter({
+      "/": {},
+    });
+    const res = await router(
+      new Request("http://localhost/"),
+    );
+
+    expect(res).toEqualResponse(
+      new Response(null, {
+        status: Status.NotFound,
+        statusText: STATUS_TEXT[Status.NotFound],
+      }),
+    );
+  },
+);
+
+it(
+  describeTests,
+  "should return 405 when method is not exists",
+  async () => {
+    const router = createRouter({
+      "/": {
+        "GET": () => new Response("hello"),
+        "HEAD": () => new Response("hello"),
+      },
+    });
+    const res = await router(
+      new Request("http://localhost/", { method: "POST" }),
+    );
+
+    expect(res).toEqualResponse(
+      new Response(null, {
+        status: Status.MethodNotAllowed,
+        statusText: STATUS_TEXT[Status.MethodNotAllowed],
+        headers: {
+          allow: "GET,HEAD",
+        },
+      }),
+    );
+  },
+);
+
+it(
+  describeTests,
+  "should return 500 when router handler has exception",
+  async () => {
+    const router = createRouter({
+      "/": () => {
+        throw Error("Unknown error");
+      },
+    });
+    const res = await router(
+      new Request("http://localhost/"),
+    );
+
+    expect(res).toEqualResponse(
+      new Response(null, {
+        status: Status.InternalServerError,
+        statusText: STATUS_TEXT[Status.InternalServerError],
+      }),
+    );
+  },
+);
+
+it(
+  describeTests,
+  "should return 200 when url path match",
+  async () => {
+    const router = createRouter({
+      "/": (_, params) => {
+        expect(params).toEqual({});
+
+        return new Response("Hello");
+      },
+    });
+    const res = await router(
+      new Request("http://localhost/"),
+    );
+
+    expect(res).toEqualResponse(
+      new Response("Hello", {
+        status: Status.OK,
+      }),
+    );
+  },
+);
+
+it(
+  describeTests,
+  "should pass params when url patten include",
+  async () => {
+    const router = createRouter({
+      "/api/:id": (_, params) => {
+        expect(params).toEqual({ id: "test" });
+        return new Response(null);
```

Comment:
Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/httpland/http-router/pullRequest?prid=9963312)
```suggestion
      return new Response(null);
```